### DISCUSSION TITLE
Livewire v3 Problem - /livewire/update 404 (Not Found)

### DISCUSSION THREAD

**Discussion by RileyGWeb** [livewire/livewire#7685] [1. Help]
I'm hesitant to call this a bug just yet, but it feels like it is. Doing a simple database query through Livewire is giving me a 404 error.

To reproduce this problem at it's simplest level, create a component with a function that performs a database query on load, and has a button that performs that query again from the component.
```
<?php

namespace App\Livewire;

use App\Models\Event;
use Livewire\Component;

class LivewireUpdateNotFoundTest extends Component
{
    public $shows;

    public function mount() {
        $this->getShows();
    }

    public function getShows() {
        $this->shows = Event::all();
    }

    public function callGetShows() {
        $this->getShows();
    }

    public function render()
    {
        return view('livewire.livewire-update-not-found-test');
    }
}
```
```
<div>
    <button wire:click="callGetShows()">Call it again</button>
</div>
```
The query is successful the first time, but upon clicking the button to call it again, I get a 404 error on my site and in the console, I get:

`POST http://127.0.0.1:8002/livewire/update 404 (Not Found)`

I checked my `php artisan route:list` and /livewire/update is there. I tried clearing my route cache (and all caches) too.

I need this functionality in my actual application, but obviously to a more complex degree. This was working before I upgraded to v3. Any ideas?

**Comment by alisalmabadi** [comment]
it could be related to the "[legacy_model_binding](https://livewire.laravel.com/docs/upgrading#eloquent-model-binding)" config which was introduced in version 3
I think it should be fixed when you make it false in livewire.php config file (as it was false in version 2!)
also, you can convert your collection to an array using the toArray() method
as it mentioned in:
[https://laracasts.com/discuss/channels/livewire/livewire-3-throwing-404-after-dispatching-event?page=1&replyId=919679](https://laracasts.com/discuss/channels/livewire/livewire-3-throwing-404-after-dispatching-event?page=1&replyId=919679)

**Reply by skyville0919** [reply]
legacy_model_binding is false at default even if I publish the config the same thing still happens and still gets livewire/update 404 not found

**Reply by AbhilashSivaraman** [reply]
> it could be related to the "[legacy_model_binding](https://livewire.laravel.com/docs/upgrading#eloquent-model-binding)" config which was introduced in version 3 I think it should be fixed when you make it false in livewire.php config file (as it was false in version 2!) also, you can convert your collection to an array using the toArray() method as it mentioned in: https://laracasts.com/discuss/channels/livewire/livewire-3-throwing-404-after-dispatching-event?page=1&replyId=919679

Can you tell detail what to change in livewire.php

**Reply by shazimali** [reply]
egacy_model_binding is false and getting same issue

**Comment by trungpv1601** [comment]
I have a same issue. For my case, The Form can not binding model when The model deleted.

So, In the Form (Livewire V3) I can not use attribute as Model. Because when I delete the Model and use the Form again.

It's will show 404 Because The Model Not Found

@calebporzio

**Reply by joshhanley** [reply]
@trungpv1601 when you delete a model, you should also clear the property, like

```php
$this->form->myModel->delete();

$this->form->myModel->null;
```

**Reply by gunnery34** [reply]
> @trungpv1601 when you delete a model, you should also clear the property, like
> 
> ```php
> $this->form->myModel->delete();
> 
> $this->form->myModel->null;
> ```

I have tried adding `$myModel->null;` after `delete()` but there is no change and still in `404 livewire/update` condition
I have created a new discussion on [#8663](https://github.com/livewire/livewire/discussions/8663) please help me

**Comment by t-prod** [comment]
Hi,

Same problem with an infinite pagination scroll. The problem occurs not with all computers or smartphones.
I'm reloading the next paginated products :

1. When I load the webpage (it's a full component) everything is OK
2. When I scroll down I call a dispatch with loadMore and load the following products

![2024-05-17 09_31_22-Terrariums 133 - Naturalys - Personnel – Microsoft_ Edge](https://github.com/livewire/livewire/assets/1824638/aa34b136-1f49-4df5-999b-cbe65b9b0b4a)

![image](https://github.com/livewire/livewire/assets/1824638/59590d13-9bd7-4b58-9cc9-fb714eddbdbc)

![image](https://github.com/livewire/livewire/assets/1824638/af84272d-003d-4ead-8553-2a9a551c40c7)

And on another computer I have this :

In my opinion, livewire/update is callef by GET which is very weird

Laravel 9.X
Livewire 3.X

I try to :

* Edit htaccess
* Set legacy_model_binding to true

Nothing works

Any ideas ?

**Reply by BVFI** [reply]
Yes I have some ideas, paste some code so that we know how you query, so that we know whats happening in the component. Depending on the difference of the issue, decide if it needs to be its own separate discussion. You cant expect a solution with just images of the network tab because there could me many issues related. I can think of 10 issues that could be causing that just on the top of my head. What do you mean "The problem occurs not with all computers or smartphones", do you mean it doesnt happen on different browsers, or it doesnt occur on responsive widths?

**Reply by atenadadkhah** [reply]
I have the exact same issue. I opened a discussion [here ](https://github.com/livewire/livewire/discussions/9085)

**Comment by t-prod** [comment]
Any ideas ?

**Reply by Muetze42** [reply]
I have the same problem. Is there finally a solution for this?
